include/libtorrent/torrent_handle.hpp

```diff
@@ -881,8 +880,9 @@ namespace aux {
 		// the piece layers are available. This function is more expensive than
 		// torrent_file() since it needs to make copies of this information.
 		//
-		// When constructing a create_torrent object from a torrent_info that's
-		// in a session, you need to use this function.
+		// The torrent_file_with_hashes() is here for backwards compatibility
+		// when constructing a create_torrent object from a torrent_info that's
+		// in a session. Prefer save_resume_data() + write_torrent_file().
```

Comment:
the `add_torrent_params` contains the trees (possibly not fully filled in) suitable for a resume file. If an `add_torrent_params` object is saved as a .torrent file (with `write_torrent_file()`) and the merkle trees aren't full (at least down to the piece layer) the call will fail with an exception `torrent_invalid_piece_layer`. However, if the flag `allow_missing_piece_layer` is passed in, it will produce a torrent without piece layers, or with incomplete ones.

Reply:
Ok 👍